resources/metadata/maven.yaml

```diff
@@ -0,0 +1,68 @@
+metadata:
+  name: mavenExecute
+  description: This step allows to run maven commands
+  longDescription: |
+    This step runs a maven command based on the parameters provided to the step.
+spec:
+  inputs:
+    params:
+      - name: pomPath
+        type: string
+        description: Path to the pom file that should be used.
+        scope:
+          - PARAMETERS
+          - STEPS
+        mandatory: false
+      - name: projectSettingsFile
+        type: string
+        description: Path to the mvn settings file that should be used as project settings file.
+        scope:
+          - PARAMETERS
+          - STEPS
+        mandatory: false
+      - name: globalSettingsFile
+        type: string
+        description: Path to the mvn settings file that should be used as global settings file.
+        scope:
+          - PARAMETERS
+          - STEPS
+        mandatory: false
+      - name: m2Path
+        type: string
+        description: Path to the location of the local repository that should be used.
+        scope:
+          - PARAMETERS
+          - STEPS
+        mandatory: false
+      - name: goals
+        type: "[]string"
+        description: Maven goals that should be executed.
+        scope:
+          - PARAMETERS
+        mandatory: true
```

Comment:
`goals` here is labeled as `mandatory`, but in case we don't provide that flag the maven build gets triggered an maven fails with a corresponding warning. I would expect the piper lib fails before calling maven in such a case. I'm almost sure that is not something inside the code of that PR, but there is something wrong in the framework --> we ave a slice here and I guess an empty slice behaves in a different way like e.g. an empty string.

Reply:
I tracked that behaviour in this issue: https://github.com/SAP/jenkins-library/issues/1216